Web/JS/create_menu copy.js

```diff
@@ -0,0 +1,182 @@
+        for (let i = 0;i < val.value.length;i++) {
+            let element_created = document.createElement("li");
+
+            let element_content = document.createElement("a");
+            element_content.onclick = show_the_article_by_category(val.value[i].header_category_id);
+            element_content.textContent = val.value[i].name;
+            let test = document.getElementById("li2" + val.value[i].header_category_id);
+            element_created.id = val.value[i].id;
+            if (val.value[i].header_category_id == -1) {
+                console.log("grande catégorie");
+
+                console.log("root" + element_created.id);
+                element_created.appendChild(element_content);
+                future_menu.appendChild(element_created);
+                console.log(future_menu);
+            } else if (document.getElementById(val.value[i].header_category_id)) {
+                //On doit créer un ul et une liste avec l'élément
+                let ul_under_root = document.createElement('ul');
+                ul_under_root.id = "ul_under_root" + val.value[i].header_category_id;
+                ul_under_root.appendChild(element_created);
+                element_created.appendChild(element_content);
+            } else if (test) {
+                //               Console.log("existe déja");
+                let li = document.getElementById("li2" + val.value[i].header_category_id);
+                let ul;
+                if (document.getElementById("li_ul" + val.value[i].header_category_id)) {
+                    //               Console.log("inside sous catégorie exise déjà");
+                    ul = document.getElementById("li_ul" + val.value[i].header_category_id);
+                    let li_inside = document.createElement("li");
+                    li_inside.id = "li_inside" + val.value[i].header_category_id;
+                    ul.appendChild(li_inside);
+                    li_inside.appendChild(element_content);
+
+                } else {
+                    console.log("inside sous catégorie exise pas");
+                    ul = document.createElement("ul");
+                    ul.id = "li_ul" + val.value[i].header_category_id;
+                    let li_inside = document.createElement("li");
+                    li_inside.id = "li_insideElse" + val.value[i].header_category_id;
+                    ul.appendChild(li_inside);
+                    li.appendChild(ul);
+                    li_inside.appendChild(element_content);
+                }
+            } else {
+                //             Console.log("sous catégorie");
+                let subCategory = document.createElement("li");
+                subCategory.id = "li" + val.value[i].header_category_id;
+                subCategory.id = "li2" + val.value[i].header_category_id;
+                subCategory.appendChild(element_content);
+                future_menu.appendChild(subCategory);
+            }
+        }
+
+        //Vide
```

Comment:
//Vide

Reply:
I don't remember what I put it. I deleted it